### PR TITLE
CJM-19966: Add fields batchInstanceID & ajoCampaignActionID in mixin Adobe CJM ExperienceEvent - Message Execution Details for Single Step Campaign

### PR DESCRIPTION
**Pull request by chavada-adobe**
## Jira
https://jira.corp.adobe.com/browse/CJM-19966

## Wiki
https://wiki.corp.adobe.com/display/CJM/One+Step+Campaign+Reporting

### CODE REVIEW

**Comment by chavada-adobe** [comment]
@prabhum2 Please review this PR.

**Comment by sawhneyh** [comment]
We should update the corresponding example file with these new fields.

**Comment by chavada-adobe** [comment]
> We should update the corresponding example file with these new fields.

@sawhneyh Added example file

**Comment by praveshg** [comment]
LGTM

File: extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json [hunk 1]
```diff
@@ -84,6 +84,16 @@
           "title": "AJO Campaign Version ID",
           "type": "string",
           "description": " Version ID of the AJO Campaign that triggered this message execution."
+        },
+        "https://ns.adobe.com/experience/customerJourneyManagement/messageExecution/ajoCampaignActionID": {
+          "title": "AJO Campaign Action ID",
+          "type": "string",
+          "description": "Action ID of the AJO Campaign that triggered this message execution."
+        },
+        "https://ns.adobe.com/experience/customerJourneyManagement/messageExecution/batchInstanceID": {
```
[the remark this answers is not in the document]
Comment (sawhneyh) [reply extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json:93]: @chetanmeh: We already have field for defining action in Journey with name JouneyAction, we can't overload that field for specifying the actions in Campaign so we created a new field for that.

Currently we don't have any field for specifying the batchInstanceId in feedback/tracking XDM. Is it fine to overload the bathInstanceId or create seprate field for both Campaign and Journey?

I think we can keep it same.

cc: @chavada-adobe

[the remark this answers is not in the document]
Comment (chetanmeh) [reply extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json:93]: Better to reuse the "neutral" batchInstanceID for both cases

**Review by sawhneyh — APPROVED** [review]

**Comment by chavada-adobe** [comment]
@cmathis @prabhum2 Please review this PR. Thanks!
cc @sawhneyh @praveshg

File: extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json [hunk 2]
```diff
@@ -84,6 +84,16 @@
...
+          "title": "Batch Instance ID for scheduled Journey or AJO Campaign Action",
```
Comment (prabhum2) [inline extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json:94]: @chavada-adobe Can this title be shortened and put description in description part?

**Comment by prabhum2** [comment]
@chavada-adobe Please let me know if this PR is ready to be reviewed and merged

**Comment by chavada-adobe** [comment]
@prabhum2 Shortened title for batchInstanceID. Now this PR is ready to be reviewed and merged.
cc @sawhneyh

**Review by prabhum2 — APPROVED** [review]

**Review by artur-ciocanu — COMMENTED** [review]
@chavada-adobe, @sawhneyh and @chetanmeh I was wondering if we have to prefix the `campaingID` and `campaginVersion` with `ajo`. We should have field names that are pretty generic. If in the future AJO will be renamed, then we will be stuck with this prefix.

**Comment by chavada-adobe** [comment]
> @chavada-adobe, @sawhneyh and @chetanmeh I was wondering if we have to prefix the `campaingID` and `campaginVersion` with `ajo`. We should have field names that are pretty generic. If in the future AJO will be renamed, then we will be stuck with this prefix.

@prabhum2 Can we mark existing fields prefixed with AJO as deprecated and create new fields without AJO prefix? If yes, what is the process to do it?
@sawhneyh @artur-ciocanu @chetanme

**Comment by prabhum2** [comment]
@chavada-adobe The two properties that you are adding as part of this PR can be edited as this is not in registry yet however if you want to deprecate any other properties that are already existing with data populated in platform you need to use "meta:status":"deprecated" at property level . Here is the example for your reference 
https://github.com/adobe/xdm/blob/master/extensions/adobe/experience/offer-management/offer-placement.schema.json

**Comment by chavada-adobe** [comment]
> @chavada-adobe The two properties that you are adding as part of this PR can be edited as this is not in registry yet however if you want to deprecate any other properties that are already existing with data populated in platform you need to use "meta:status":"deprecated" at property level . Here is the example for your reference https://github.com/adobe/xdm/blob/master/extensions/adobe/experience/offer-management/offer-placement.schema.json

@prabhum2 Thanks for sharing this. I have updated PR accordingly. 
@prabhum2  @sawhneyh @chetanmeh @artur-ciocanu Please review.

**Review by sawhneyh — APPROVED** [review]

**Review by artur-ciocanu — APPROVED** [review]
Awesome!

**Review by prabhum2 — APPROVED** [review]

**Pull request merged**